File: docs/guides/dart/lib/enableFilterPromote.dart
import 'package:algolia_client_search/algolia_client_search.dart';

void enableFilterPromote() async {
  final condition = Condition(
    pattern: "{facet:brand}",
    anchoring: Anchoring.is_,
  );

  final consequence = Consequence(filterPromotes: true);

  // ignore: unused_local_variable
  final rule = Rule(
      objectID: "rule_with_filterPromotes",
      conditions: [condition],
      consequence: consequence);
}

File: docs/guides/scala/src/main/scala/enableFilterPromote.scala
import scala.concurrent.ExecutionContext.Implicits.global

import algoliasearch.api.SearchClient
import algoliasearch.config.*
import algoliasearch.extension.SearchClientExtensions
import algoliasearch.search.{Anchoring, Condition, Consequence, Rule}

def enableFilterPromote(): Unit = {
  val condition = Condition(
    pattern = Some("{facet:brand}"),
    anchoring = Some(Anchoring.Is)
  )

  val consequence = Consequence(
    filterPromotes = Some(true)
  )

  val rule = Rule(
    objectID = "rule_with_filterPromotes",
    conditions = Some(Seq(condition)),
    consequence = consequence
  )
}
